### ISSUE TITLE
[regression/8.0.0] [Android] "SwipeItem" "CommandParameter" fails to bind as null in the CollectionView

### ISSUE THREAD

**Issue by MadCode1982** [dotnet/maui#17644]
### Description

C#

`public ICommand DeleteCommand => new Command<Monkey>(DeleteMonkey);`
`void DeleteMonkey(Monkey monkey)`
`{`
`    if (Monkeys.Contains(monkey))`
`       Monkeys.Remove(monkey);`
`}`

XAML

`<SwipeItem Text="Delete"`
               `BackgroundColor="LightPink"`
               `Command="{Binding Source={x:Reference collectionView}, Path=BindingContext.DeleteCommand}"`
               `CommandParameter="{Binding}" />`

When executed, 'monky' is null.

### Link to public reproduction project repository

https://github.com/MadCode1982/SwipeItemErrorDemo/

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.49

### Affected platforms

Android

### Did you find any workaround?

Not

**Comment by PureWeen** [comment]
This should be fixed in RC2

I tested your sample against our nightly feed and the monkeys are now coming through https://github.com/dotnet/maui/wiki/Nightly-Builds

**Comment by marinaandricic** [comment]
when will   RC2 be published

**Comment by MauiUIui** [comment]
> when will RC2 be published

Exactly. When? Please someone answer!
Such bugs are very hard to workaround or even replace with similar behaviour for deleting, especially when MAUI to this very day doesn't have support for a long press gesture!

**Comment by MauiUIui** [comment]
Oh yea, speaking of nightly, I just tried the "8-0-100-rc-2-23502-2-transport-signed-1" and it brought back the corner radius issue of the button when the Background property is set. This was actually fixed in the RC 1

**Comment by aabc123** [comment]
> This should be fixed in RC2
> 
> I tested your sample against our nightly feed and the monkeys are now coming through https://github.com/dotnet/maui/wiki/Nightly-Builds

I've just tried our app with 8.0.0-nightly.9370 and still getting null for the CommandParameter.

**Comment by MauiUIui** [comment]
> I've just tried our app with 8.0.0-nightly.9370 and still getting null for the CommandParameter.

That doesn't sound promising :/
Where did you get that nightly from and how did you use it? This [link](https://github.com/dotnet/maui/wiki/Nightly-Builds) doesn't seem to have nightly builds

**Comment by MauiUIui** [comment]
> when will RC2 be published

Now it's official. It's coming on Tuesday, 10th.

**Comment by aabc123** [comment]
> > I've just tried our app with 8.0.0-nightly.9370 and still getting null for the CommandParameter.
> 
> That doesn't sound promising :/ Where did you get that nightly from and how did you use it? This [link](https://github.com/dotnet/maui/wiki/Nightly-Builds) doesn't seem to have nightly builds

You need a nuget.config which adds https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json as a package source then change your PackageReference version for Microsoft.Maui.Controls to 8.0.0-nightly.9370

I'm assuming that's all you need to do.  Maybe I've missed a step and that's why it didn't work for me.  (This is with my app I'm currently porting from Xamarin, not with the example posted here).

**Comment by aabc123** [comment]
> > when will RC2 be published
> 
> Now it's official. It's coming on Tuesday, 10th.

I'm going to wait and try with that.  I've got our app running under Maui 7 and 8RC1 but both have different issues so still don't have a working replacement for Xamarin.Forms.

**Comment by MauiUIui** [comment]
> 8.0.0-nightly.9370

But where do yo find out what the latest nightly build is? Where did you get the version?

**Comment by aabc123** [comment]
> But where do yo find out what the latest nightly build is? Where did you get the version?

On the nightly build page PureWeen posted a link to, there's a link to the nightly builds: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8

Search in there for Maui

**Comment by aabc123** [comment]
https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet8/NuGet/Microsoft.Maui.Controls/versions/8.0.0-rc.1.9171%20sha.269a8f03ec-azdo.8365272

**Comment by MauiUIui** [comment]
> Search in there for Maui

Wanna know what sucks? It's the azure search function! I've been searching under the link you posted for a while now. I always searched for "nightly" or "night" which yielded nothing!

**Comment by MauiUIui** [comment]
> > This should be fixed in RC2
> > I tested your sample against our nightly feed and the monkeys are now coming through https://github.com/dotnet/maui/wiki/Nightly-Builds
> 
> I've just tried our app with 8.0.0-nightly.9370 and still getting null for the CommandParameter.

So hey there once more. I have good news for you. I used version "8.0.0-nightly.9370+sha.d02f597f04-azdo.8493478" and the bug is gone. The CommandParameter returns the binded value as expected.

**Comment by PureWeen** [comment]
@MauiUIui do you have a repro I can test?

The repro that's part of this issue works fine for me. 

Can you open a new issue with a reproduction please?

**Comment by PureWeen** [comment]
> > This should be fixed in RC2
> > I tested your sample against our nightly feed and the monkeys are now coming through https://github.com/dotnet/maui/wiki/Nightly-Builds
> 
> I've just tried our app with 8.0.0-nightly.9370 and still getting null for the CommandParameter.

Do you have a repro? I've tested this using the authors original PR and it works fine for me there.

**Comment by aabc123** [comment]
> 
> Do you have a repro? I've tested this using the authors original PR and it works fine for me there.

I don't know how to tell if it was correctly linking against the nightly.  So I'll test again when RC2 is out and post back.  It's in a very large app I'm porting from Xamarin.Forms so can't post it as is but if it doesn't work, I'll try and cut it down to a sample app and post here.

Thanks.

**Comment by PureWeen** [comment]
> > Do you have a repro? I've tested this using the authors original PR and it works fine for me there.
> 
> I don't know how to tell if it was correctly linking against the nightly. So I'll test again when RC2 is out and post back. It's in a very large app I'm porting from Xamarin.Forms so can't post it as is but if it doesn't work, I'll try and cut it down to a sample app and post here.
> 
> Thanks.

If you get it down to a really simple repro can you create a new issue?

Can you just post your use of SwipeView and the CollectionView it's part of? This way I can see how your bindings are setup? 

What platform are you on?

**Comment by aabc123** [comment]
> 
> If you get it down to a really simple repro can you create a new issue?
> 
> Can you just post your use of SwipeView and the CollectionView it's part of? This way I can see how your bindings are setup?
> 
> What platform are you on?

It's running on iOS 17.  Works with Maui 7.0.92.  It's actually a scroll view containing a vertical stack layout using BindableLayout for its items.  The data template is a vertical stack layout containing a SwipeViewItem as the single right item.  Perhaps the issue is with SwipeItemView?

**Comment by PureWeen** [comment]
> > If you get it down to a really simple repro can you create a new issue?
> > Can you just post your use of SwipeView and the CollectionView it's part of? This way I can see how your bindings are setup?
> > What platform are you on?
> 
> It's running on iOS 17. Works with Maui 7.0.92. It's actually a scroll view containing a vertical stack layout using BindableLayout for its items. The data template is a vertical stack layout containing a SwipeViewItem as the single right item. Perhaps the issue is with SwipeItemView?

If you change the BindableLayout to a CollectionView do you see the same issue?

**Comment by aabc123** [comment]
> If you change the BindableLayout to a CollectionView do you see the same issue?

I'm still seeing the issue with CollectionIVew in my usage with 8.0 RC1.

**Comment by MauiUIui** [comment]
> > > If you get it down to a really simple repro can you create a new issue?
> > > Can you just post your use of SwipeView and the CollectionView it's part of? This way I can see how your bindings are setup?
> > > What platform are you on?
> > 
> > 
> > It's running on iOS 17. Works with Maui 7.0.92. It's actually a scroll view containing a vertical stack layout using BindableLayout for its items. The data template is a vertical stack layout containing a SwipeViewItem as the single right item. Perhaps the issue is with SwipeItemView?
> 
> If you change the BindableLayout to a CollectionView do you see the same issue?

Hi there. Sorry for taking some time to reply.
I've been trying for an hour to repro but couldn't. Yesterday it was happening about 60 % of the time. I didn't change a thing here since yesterday but did restart my Windows ;) 
Anyway if it helps, with me it's a little bit (more) complicated. It's happening on Android. Here it's a `ListView` whose `ItemTemplate` is a `DataTemplateSelector` and in one of it's templates (mostly the only one that ever gets selected) there is a `ViewCell` (mandatory for the `ListView`) which has the `SwipeView` in it. This in return has one `SwipeItem` in its `LeftItems`. Here lies the `Command` and of course the `CommandParameter`.
AGAIN: The problem is (was??!) with the `Command` itself (its `Execute` method doesn't fire). But whenever it does fire, the `CommandParameter` is actually there (only after trying with the previously mentioned Nightly).
Binding the command from inside the `SwipeItem` looks something like this:
`Command="{Binding BindingContext.DeleteItemCommand, Source={x:Reference This}}" CommandParameter="{Binding}"`
"This" would be the name of the `ContentView`.
Bear in mind that I'm implementing MVVM with **Compiled** bindings. I also **have to** set the DataType of the DataTemplate to null like this `<DataTemplate x:DataType="{x:Null}">` since otherwise the BindingContext of the `DataTemplate` doesn't inherit form the templated item (I guess this one is another bug I still have to live with for a while ;)).

**Comment by aabc123** [comment]
> UPDATE: I got it reproduced by building a release. It happened both on a physical device and on the emulator. I had it happen when I had less than a certain number of items in the collection (Sometimes 4, other times 6). If I keep adding new items to the collection it just acts as expected. Memory leak?!

Thanks.  My stack layout is a custom control that is re-used and displayed in the middle of a scroll view along with other controls (so CollectionView doesn't really work for me).  Typically there is only 1-3 items in my bindable items list.  Here's the XAML:

`<?xml version="1.0" encoding="UTF-8" ?>
<StackLayout
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    x:Class="Controls.GoalsList"
    x:Name="_control"
    Style="{StaticResource STACK_LAYOUT_LIST_STYLE}">

    <StackLayout.Resources>
        <DataTemplate x:Key="_maintainDataTemplate" x:DataType="vm:GoalViewModel">
            <VerticalStackLayout BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}"
                                 Padding="{x:Static R:Theme.LR_THICKNESS}">
                <SwipeView IsEnabled="{Binding IsReadOnly, Converter={x:Static R:Theme.NEGATE_CONVERTER}}"
                           BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}"
                           Padding="{x:Static R:Theme.TB_QUARTER_THICKNESS}">
                    <SwipeView.RightItems>
                        <c:SwipeItem Text="{x:Static str:Strings.SWIPE_BUTTON_DELETE}" Style="{StaticResource SWIPE_ITEM_DESTRUCTIVE_STYLE}"
                                     LeftSpacing="{x:Static R:Theme.QUARTER_SPACING}"
                                     Command="{Binding DeleteCommand, Source={x:Reference _control}}"
                                     CommandParameter="{Binding }" />
                    </SwipeView.RightItems>
                    <c:MaintainGoalView ShowMetric="{Binding ShowMetric, Source={x:Reference _control}}"
                                        BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}">
                        <c:MaintainGoalView.GestureRecognizers>
                            <TapGestureRecognizer Command="{Binding EditCommand, Source={x:Reference _control}}" CommandParameter="{Binding }" />
                        </c:MaintainGoalView.GestureRecognizers>
                    </c:MaintainGoalView>
                </SwipeView>
            </VerticalStackLayout>
        </DataTemplate>

        <DataTemplate x:Key="_gainLoseDataTemplate" x:DataType="vm:GoalViewModel">
            <VerticalStackLayout BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}"
                                 Padding="{x:Static R:Theme.LR_THICKNESS}">
                <SwipeView IsEnabled="{Binding IsReadOnly, Converter={x:Static R:Theme.NEGATE_CONVERTER}}"
                           BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}"
                           Padding="{x:Static R:Theme.TB_QUARTER_THICKNESS}">
                    <SwipeView.RightItems>
                        <c:SwipeItem Text="{x:Static str:Strings.SWIPE_BUTTON_DELETE}" Style="{StaticResource SWIPE_ITEM_DESTRUCTIVE_STYLE}"
                                     LeftSpacing="{x:Static R:Theme.QUARTER_SPACING}"
                                     Command="{Binding DeleteCommand, Source={x:Reference _control}}"
                                     CommandParameter="{Binding }" />
                    </SwipeView.RightItems>
                    <c:GainLoseGoalView ShowMetric="{Binding ShowMetric, Source={x:Reference _control}}"
                                        BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}">
                        <c:GainLoseGoalView.GestureRecognizers>
                            <TapGestureRecognizer Command="{Binding EditCommand, Source={x:Reference _control}}" CommandParameter="{Binding }" />
                        </c:GainLoseGoalView.GestureRecognizers>
                    </c:GainLoseGoalView>
                </SwipeView>
            </VerticalStackLayout>
        </DataTemplate>
    </StackLayout.Resources>

    <BindableLayout.ItemTemplateSelector>
        <c:GoalViewTemplateSelector Maintain="{StaticResource _maintainDataTemplate}" GainLose="{StaticResource _gainLoseDataTemplate}" />
    </BindableLayout.ItemTemplateSelector>

    <BindableLayout.EmptyView>
        <Label Text="{x:Static str:Strings.GOALS_VIEW_NO_GOAL_SET_TEXT}" HorizontalOptions="Fill" HorizontalTextAlignment="Center"
                BackgroundColor="{Binding ListItemBackgroundColour, Source={x:Reference _control}}">
            <Label.Padding>
                <Thickness Top="{x:Static R:Theme.QUARTER_SPACING}" Bottom="{x:Static R:Theme.QUARTER_SPACING}" />
            </Label.Padding>
        </Label>
    </BindableLayout.EmptyView>
</StackLayout>`

**Comment by aabc123** [comment]
Except the editor has stripped off the top XAML entry and closing tag for some reason.

**Comment by aabc123** [comment]
Just to confirm, my issue is fixed with 8.0 rc2.

**Comment by MauiUIui** [comment]
> Just to confirm, my issue is fixed with 8.0 rc2.

Mine as well (it actually only existed in RC1, the nightly build actually already had fixed it). Thanks
